server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -639,102 +639,115 @@ public final void writeOptionalInstant(@Nullable Instant instant) throws IOExcep
-        o.writeVInt(bytes.length);
-        o.writeBytes(bytes);
-    }), entry(List.class, (o, v) -> {
-        o.writeByte((byte) 7);
-        final List<?> list = (List<?>) v;
-        o.writeVInt(list.size());
-        for (Object item : list) {
-            o.writeGenericValue(item);
-        }
-    }), entry(Object[].class, (o, v) -> {
-        o.writeByte((byte) 8);
-        final Object[] list = (Object[]) v;
-        o.writeVInt(list.length);
-        for (Object item : list) {
-            o.writeGenericValue(item);
-        }
-    }), entry(Map.class, (o, v) -> {
-        if (v instanceof LinkedHashMap) {
-            o.writeByte((byte) 9);
-        } else {
-            o.writeByte((byte) 10);
-        }
-        @SuppressWarnings("unchecked")
-        final Map<String, Object> map = (Map<String, Object>) v;
-        o.writeVInt(map.size());
-        for (Map.Entry<String, Object> entry : map.entrySet()) {
-            o.writeString(entry.getKey());
-            o.writeGenericValue(entry.getValue());
-        }
-    }), entry(Byte.class, (o, v) -> {
-        o.writeByte((byte) 11);
-        o.writeByte((Byte) v);
-    }), entry(Date.class, (o, v) -> {
-        o.writeByte((byte) 12);
-        o.writeLong(((Date) v).getTime());
-    }), entry(BytesReference.class, (o, v) -> {
-        o.writeByte((byte) 14);
-        o.writeBytesReference((BytesReference) v);
-    }), entry(Text.class, (o, v) -> {
-        o.writeByte((byte) 15);
-        o.writeText((Text) v);
-    }), entry(Short.class, (o, v) -> {
-        o.writeByte((byte) 16);
-        o.writeShort((Short) v);
-    }), entry(int[].class, (o, v) -> {
-        o.writeByte((byte) 17);
-        o.writeIntArray((int[]) v);
-    }), entry(long[].class, (o, v) -> {
-        o.writeByte((byte) 18);
-        o.writeLongArray((long[]) v);
-    }), entry(float[].class, (o, v) -> {
-        o.writeByte((byte) 19);
-        o.writeFloatArray((float[]) v);
-    }), entry(double[].class, (o, v) -> {
-        o.writeByte((byte) 20);
-        o.writeDoubleArray((double[]) v);
-    }), entry(BytesRef.class, (o, v) -> {
-        o.writeByte((byte) 21);
-        o.writeBytesRef((BytesRef) v);
-    }), entry(GeoPoint.class, (o, v) -> {
-        o.writeByte((byte) 22);
-        o.writeGeoPoint((GeoPoint) v);
-    }), entry(ZonedDateTime.class, (o, v) -> {
-        o.writeByte((byte) 23);
-        final ZonedDateTime zonedDateTime = (ZonedDateTime) v;
-        o.writeString(zonedDateTime.getZone().getId());
-        o.writeLong(zonedDateTime.toInstant().toEpochMilli());
-    }), entry(Set.class, (o, v) -> {
-        if (v instanceof LinkedHashSet) {
-            o.writeByte((byte) 24);
-        } else {
-            o.writeByte((byte) 25);
-        }
-        o.writeCollection((Set<?>) v, StreamOutput::writeGenericValue);
-    }),
+    private static final Map<Class<?>, Writer<?>> WRITERS = Map.ofEntries(
+        entry(String.class, (o, v) -> o.writeStringGeneric((String) v)),
```

Comment:
Sorry for this noise, couldn't find a way to make spotless not do this.

Reply:
Seems weird to selectively suppress that here just to save on the diff? The old formatting was kind of strange to begin with I guess.